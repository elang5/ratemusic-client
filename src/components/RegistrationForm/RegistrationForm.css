.reg-form {
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
  text-align: center;
  width: 65%;
  height: 62%;
  border-radius: 5.5px;
  box-shadow: 0 0 3px 2px rgba(0, 0, 0, 0.15);
  background-color: #ffffff;
}

.error {
  color: red;
  font-size: 14px;
}

.register-text {
  color: #1f1f21;
  font-size: 24px;
}

/* fix label global classname */

#user_name, #full_name {
  margin-bottom: .5em;
}

#user_name, #password, #full_name {
  margin-top: .5em;
  width: 41.5%;
  padding: 15px 20px;
  border-radius: 2.2px;
  border: solid 0.6px rgba(208, 208, 208, 0.5);
  background-color: #e5e9ed;
  font-size: 20px;
}


.register-btn {
  width: 41.5%;
  padding: 15px 20px;
  background-color: rgb(252, 60, 57);
  margin: 1em 0;
  border-radius: 2.2px;
  font-size: 20px;
  color: white;
  font-weight: 200;
}