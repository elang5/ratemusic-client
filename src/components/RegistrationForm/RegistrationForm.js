import React, { Component } from 'react'
import AuthApiService from '../../services/auth-api-service'
import './RegistrationForm.css'

export class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }

  state = { error: null }

  handleSubmit = e => {
    e.preventDefault()
    const { full_name, user_name, password } = e.target

    AuthApiService.postUser({
      user_name: user_name.value,
      password: password.value,
      full_name: full_name.value
    })
      .then(user => {
        user_name.value = ''
        password.value = ''
        full_name.value = ''
        this.props.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }
  
  render() {
    const { error } = this.state
    return (
      <form 
        className="reg-form"
        onSubmit={this.handleSubmit}>
        <div className="error">
          {error && <p className="error">{error}</p>}
        </div>
        <p className="register-text">Register for Rate Music</p>
        <div className="full_name">
          <label className="label" htmlFor="full_name">Full name </label><br />
          <input 
            required
            type="text" 
            name="full_name"
            id="full_name"/>
        </div>
        <div className="user_name">
          <label className="label" htmlFor="user_name">Username</label><br />
          <input 
            required
            type="text"
            name="user_name"
            id="user_name"
          />
        </div>
        <div className="password">
          <label className="label" htmlFor="password">Password</label>
          <p className="password-rules">Please create an 8+ character password with a number, uppercase letter, and special character.</p>
          <input 
            required
            type="password"
            name="password"
            id="password"
            />
        </div>
        <div className="container">
          <button type="submit" className="register-btn">
            Register
          </button>
        </div>
      </form>
    )
  }
}

export default RegistrationForm
