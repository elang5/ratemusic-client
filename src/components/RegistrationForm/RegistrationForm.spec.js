import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RegistrationForm from './RegistrationForm'

describe(`RegistrationForm Component`, () => {

  it(`renders the RegistrationForm component`, () => {
    const wrapper = shallow(<RegistrationForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})