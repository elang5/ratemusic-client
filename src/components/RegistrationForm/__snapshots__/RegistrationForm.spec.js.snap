// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm Component renders the RegistrationForm component 1`] = `
<form
  className="reg-form"
  onSubmit={[Function]}
>
  <div
    className="error"
  />
  <p
    className="register-text"
  >
    Register for Rate Music
  </p>
  <div
    className="full_name"
  >
    <label
      className="label"
      htmlFor="full_name"
    >
      Full name 
    </label>
    <br />
    <input
      id="full_name"
      name="full_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      className="label"
      htmlFor="user_name"
    >
      User name
    </label>
    <br />
    <input
      id="user_name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      className="label"
      htmlFor="password"
    >
      Password
    </label>
    <br />
    <input
      id="password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="container"
  >
    <button
      className="register-btn"
      type="submit"
    >
      Register
    </button>
  </div>
</form>
`;
